src/helpers/domain.js

```diff
@@ -160,26 +162,51 @@ export default {
     return categories.length === 0 ? _dataByIndex() : _dataByCategory();
   },
 
-  padDomain(domain, props, axis) {
+  padDomain(domain, props, axis) { // eslint-disable-line max-statements
     if (!props.domainPadding) {
       return domain;
     }
-    const domainPadding = typeof props.domainPadding === "number" ?
-      props.domainPadding : props.domainPadding[axis];
 
-    if (!domainPadding) {
-      return domain;
-    }
+    let adjustedMax;
+    let adjustedMin;
     const domainMin = Math.min(...domain);
     const domainMax = Math.max(...domain);
     const range = Helpers.getRange(props, axis);
     const rangeExtent = Math.abs(Math.max(...range) - Math.min(...range));
-    const padding = Math.abs(domainMax - domainMin) * domainPadding / rangeExtent;
-    // don't make the axes cross if they aren't already
-    const adjustedMin = (domainMin >= 0 && (domainMin - padding) <= 0) ?
-      0 : domainMin.valueOf() - padding;
-    const adjustedMax = (domainMax <= 0 && (domainMax + padding) >= 0) ?
-      0 : domainMax.valueOf() + padding;
+
+    if (typeof props.domainPadding !== "number"
```

Comment:
I would extract all of this into a `getDomainPadding` method that takes `props` and `axis` returns `{left: NUMBER, right: NUMBER}` or something like that